### ISSUE TITLE
Error when different users create folders with the same name

### ISSUE THREAD

**Issue by juanibarral** [donhinkelman/moodle-block_sharing_cart#95]
**Description**
When two different users save activities in folders with the same name, copying any of those folders to a section will result on an exception:
![image](https://user-images.githubusercontent.com/3487934/112903598-58e3d400-90ad-11eb-801c-f8d59368855e.png)

**Replication steps**
Log in as a teacher/admin (user1)
Creates an activity in any course
Send the activity to sharing cart
Send the activity to a folder named **testfolder** using the **"move to folder"** icon
Log out
Log in as another teacher/admin (user2)
Creates an activity in any other course
Send the activity to sharing cart
Send the activity to a folder named **testfolder** using the **"move to folder"** icon
Send the folder to a section using the **"Copy to course"** icon
Select the target section 

**Observed**
An exception is shown

**Expected**
The activities inside the folder are pasted in the selected section

I created this patch for this issue:

[fix.txt](https://github.com/donhinkelman/moodle-block_sharing_cart/files/6224758/fix.txt)

**Comment by donhinkelman** [comment]
Hello Juan, thank you for reporting this issue #95. I will assign it to Ponlawat who originally programmed the section copy feature of the Sharing Cart.

**Comment by ponlawat-w** [comment]
Hi,

I am trying to reproduce the problem with the latest version of Sharing Cart, and found that the problem has already been solved.
Looking to your attached patch file, it seems that there is a previous commit already fix the issue:
https://github.com/donhinkelman/moodle-block_sharing_cart/commit/382b476f76b25b714c674dc76ddc65361f134a08

I also added user ID condition of `$items` in `restore.php`, which is not related to the issue, but it was proposed by your patch (pull request #99). That part is used in warning message of heavy restore message.

Thank you,
